### PR TITLE
fix: add yaml validations in training model

### PR DESCRIPTION
**Pull request by shekhar-rudder**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/py_native/training.py [hunk 1]
```diff
@@ -76,9 +76,16 @@ def get_material_recipe(self) -> PyNativeRecipe:
         return TrainingRecipe(self.build_spec)
 
     def validate(self) -> tuple[bool, str]:
-        min_version = constants.MIN_PB_VERSION
-        if self.schema_version < min_version:
-            return False, f"schema version should >= {min_version}"
+        data_config = self.build_spec["ml_config"]["data"]
+        task = data_config.get("task", "")
+        if task != "regression" and task != "":
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/training.py:81]: Task can be `classification` or `regression` both are valid. 
If it is blank, we impute that to classification.

File: src/predictions/profiles_mlcorelib/py_native/training.py [hunk 2]
```diff
@@ -76,9 +76,16 @@ def get_material_recipe(self) -> PyNativeRecipe:
...
+            return (
+                False,
+                f"Invalid task {task}. Supported tasks - classification, regression",
+            )
+        label_value = data_config.get("label_value", "")
+        if label_value == "" and task == "regression":
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/training.py:87]: Label value should not be present for regression, so it should be the other way - if task == "regression" and label_value!="" then return False. 

For classification, label_value is strongly recommended but if not given, we have an inference logic internally. So if label_value is not given, we don't need to raise an exception.

File: samples/py_native/models/profiles.yaml [hunk 3]
```diff
@@ -33,6 +33,7 @@ models:
         data:
           task: regression
           label_column: days_since_last_seen
+          label_value: 1
```
Comment (dpatchigolla) [inline samples/py_native/models/profiles.yaml:36]: regression and label_value should throw an exception.

**Review by dpatchigolla — CHANGES_REQUESTED** [review]

**Comment by dpatchigolla** [comment]
Can you also add label_value to classification model in the pynative sample yaml? you can give value = 1.

File: samples/py_native/models/profiles.yaml (hunk 3 above)
Comment (shekhar-rudder) [reply samples/py_native/models/profiles.yaml]: Removed from regression YAML

File: src/predictions/profiles_mlcorelib/py_native/training.py (hunk 1 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/training.py]: Made `classification` a valid value

File: src/predictions/profiles_mlcorelib/py_native/training.py (hunk 2 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/training.py]: Updated the check

**Comment by shekhar-rudder** [comment]
> Can you also add label_value to classification model in the pynative sample yaml? you can give value = 1.

Done

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**